pkg/controller/control.go

```diff
@@ -58,12 +61,16 @@ type Controller struct {
 	lastExpansionError string
 
 	fileSyncHTTPClientTimeout int
+
+	profilerServer *http.Server
+	profilerLock   sync.Mutex
 }
 
 const (
 	freezeTimeout         = 60 * time.Minute // qemu uses 60 minute timeouts for freezing
 	syncTimeout           = 60 * time.Minute
 	lastModifyCheckPeriod = 5 * time.Second
+	profilerPortBase      = 20001
```

Comment:
NIT: add a comment to explain why

Reply:
I added a comment and moved to `pkg/types/types.go` to avoid duplicates. Thanks!